base/exports.jl

```diff
@@ -1430,6 +1430,7 @@ export
     @inbounds,
     @fastmath,
     @simd,
+    @shareindexes,
```

Comment:
FYI re #12902 I'm OK with either outcome. I was originally surprised by `indexes` but have grown fond of it.

Since it looks like the closure solution might not be feasible for a while, we might need the macro. Should I rename this `@shareindex` simply to circumvent the debate?